pkg/providers/placementgroup/placementgroup.go

```diff
@@ -0,0 +1,94 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placementgroup
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+
+	"github.com/aws/karpenter-core/pkg/utils/pretty"
+	"github.com/samber/lo"
+	"knative.dev/pkg/logging"
+)
+
+type Provider struct {
+	sync.RWMutex
+	ec2api ec2iface.EC2API
+	cache  *cache.Cache
+	cm     *pretty.ChangeMonitor
+}
+
+func NewProvider(ec2api ec2iface.EC2API, cache *cache.Cache) *Provider {
+	return &Provider{
+		ec2api: ec2api,
+		cm:     pretty.NewChangeMonitor(),
+		// TODO: Remove cache for v1beta1, utilize resolved subnet from the AWSNodeTemplate.status
+		// Subnets are sorted on AvailableIpAddressCount, descending order
+		cache: cache,
+	}
+}
+
+func (p *Provider) Get(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (*ec2.PlacementGroup, error) {
```

Comment:
From looking at the code, I see that we have a `placementGroupSelector` but we aren't actually using this in the launch template when we generate the placement. Should we be using this as well?

Reply:
`selectors := nodeClass.Spec.LicenseSelectorTerms` should have been `selectors := nodeClass.Spec.placementGroupSelector`.
fixed.